### PR TITLE
Embeds admin web app into CLI with a yarn command for development

### PR DESCRIPTION
**Pull request by mitchhentgesspotify**
* Added an `admin-web` command to the CLI, but the web app isn't currently bundled properly for distribution so it's a no-op.
* `yarn start-admin-web` allows local development of this very simple admin web app
  * There's some hackery in `local-dev.ts` to make `serveBundle()` work in this new use case - I tried to make the minimal config that would satisfy the type system and work.
* Adds some example HTML, CSS and JS as demonstration for how to use each

image

### CODE REVIEW

**Comment by andrewthauer** [comment]
I'm curious, what are the intended use cases of this?

**Comment by mitchhentgesspotify** [comment]
> I'm curious, what are the intended use cases of this?

Good question: my team wants to address the sharp edges of Backstage adoption by providing a simplified "admin page" developer tool that enables a "guided" setup of Backstage (such as setting up authentication with GitHub, integrating with the catalog plugin, and other commonly-used features).

We'll likely prepare an RFC for this once some more unknowns slide into place, which should make the plans here more clear.

**Comment by awanlin** [comment]
@mitchhentgesspotify very excited to see this even though I'm pretty far into my Backstage journey I help a lot of people on Discord (I'm `@Ahhhndre`) and I think this will really reduce some of the questions we see there.

I mentioned this to Francesco but this really reminds me of the VueJS CLI GUI and you might want to take some inspiration from there as it's pretty beautiful with a great UX/DX IMO - https://medium.com/@brandonl.4231/getting-to-know-the-new-vue-cli-3-user-interface-a173b00128bd

**Review by samiramkr — APPROVED** [review]

File: packages/cli/src/commands/admin-web/local-dev.ts
Comment (djamaile) [inline packages/cli/src/commands/admin-web/local-dev.ts:21]: Testing left over?

File: packages/cli/src/commands/admin-web/local-dev.ts
Comment (djamaile) [inline packages/cli/src/commands/admin-web/local-dev.ts:34]: nit: we can just run this on `3000` right? Either way port 7008 is a bit weird out of place for a web uit? Or maybe I am looking to deep into this.

File: packages/cli/src/commands/admin-web/embedded/public/index.html
Comment (djamaile) [inline packages/cli/src/commands/admin-web/embedded/public/index.html:1]: What was the decision between having a minimum React app embedded instead of plain HTML? I agree that might've make the PR a bit too big(?).

**Review by djamaile — APPROVED** [review]
🆒 beans

File: packages/cli/src/commands/admin-web/local-dev.ts
Comment (mitchhentgesspotify) [reply packages/cli/src/commands/admin-web/local-dev.ts:34]: `3000` is cool, but Backstage (I guess backend, heh) runs on `7007`, and I figured that we _probably_ don't want to conflict, "just in case"? I'm not attached to `7008` though :)

File: packages/cli/src/commands/admin-web/embedded/public/index.html
Comment (mitchhentgesspotify) [reply packages/cli/src/commands/admin-web/embedded/public/index.html:1]: I believe that'll setting up React will be tackled in [this ticket](https://github.com/orgs/backstage/projects/5/views/1?pane=issue&itemId=19983621).

I wanted to keep this PR small and tight to unblock work that depended on it (for the same reason that I haven't made "dist builds" work with this yet)

**Comment by djamaile** [comment]
@mitchhentgesspotify btw don't forget to sign your commits it might become a hassle once we want to merge to main.

**Comment by mitchhentgesspotify** [comment]
I've figured that there may be some rebasing and "cleaning up" (as well as changeset management, etc) once we reach a minimum-viable-mergeable state, but I agree that there's no harm in sign-off-ing commits in advance :+1:

**Pull request merged**
